### ISSUE TITLE
Responsive flat accessibility issue

### ISSUE THREAD

**Issue by edds** [Financial-Times/o-table#180]
## What

The duplicate headings that the responsive flat table adds in are causing our Pa11y tests to fail with the following message: 

> The relationship between td elements and their associated th elements is not
> defined. As this table has multiple levels of th elements, you must use the
> headers attribute on td elements.

## Details

This can be seen on the [responsive flat table demo](https://www.ft.com/__origami/service/build/v2/demos/o-table@8.0.1/responsive-flat?brand=master), the multiple headers that can apply to the `td` tags means assistive tech doesn't know which headers to use. 

```html
<tr data-o-table-flat-headings="true">
	<th scope="row" role="rowheader" class="o-table__duplicate-heading">Fruit</th><td>Dragonfruit</td>
	<th scope="row" role="rowheader" class="o-table__duplicate-heading">Genus</th><td>Stenocereus</td>
	<th scope="row" role="rowheader" class="o-table__duplicate-heading">Characteristic</th><td>Juicy</td>
	<th scope="row" role="rowheader" data-o-table-data-type="numeric" class="o-table__cell--numeric o-table__duplicate-heading">Cost&nbsp;(GBP)</th><td class="o-table__cell--numeric">3</td>
	<th scope="row" role="rowheader" data-o-table-data-type="numeric" class="o-table__cell--numeric o-table__duplicate-heading">Cost&nbsp;(EUR)</th><td class="o-table__cell--numeric">2.72</td>
</tr>
```

The relevant WACG technique page for this: [Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H43.html).

**Comment by notlee** [comment]
Hmmm I'm not sure about this one. I think it might be unnecessary and would rather not calculate the `headers` attribute for each cell if we can help it. The duplicate headings are created with JS and hidden from screen readers using `display: none`. On a mobile breakpoint the headings switch and the original headings are hidden with `display: none` instead. So there are only ever one set of headings visible (visually and to screen readers) which also use the `scope` attribute. So there shouldn't be ambiguity -- unless JS executes and CSS fails to load at least 😬 

Questions so far:
- Are the hidden headers actually visible to screen readers after all; or
- Is Pa11y looking at markup only and thus doesn't count duplicate headers as hidden; or
- Is Pally running against a page with no CSS

I added the scroll and flat versions to our own pa11y test and got no errors so need to look into why

**Comment by notlee** [comment]
Good news! Our demo failed in the same way in Circle -- I must have been using an older version of pa11y. https://circleci.com/gh/Financial-Times/o-table/596?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

**Comment by edds** [comment]
On a mobile layout is exactly when the markup doesn't make sense . The markup which I listed above all gets rendered with no `display: none`. In that situation you have 5 headers on one row which are all claiming to be the `rowheader`. They might visually they look like different rows but assistive tech it can't see that.

**Comment by notlee** [comment]
Oh I see! Sorry I'm not sure where my brain went with this. Thanks for explaining again. We'll put together a fix 😊

**Comment by notlee** [comment]
A fix for this was released in v8.0.3. The table markup is now valid on mobile 
devices; it is no longer announced as an empty table using VoiceOver; and 
is in our own pa11y test. But there is room for improvement regarding how 
each item in the table is delineated to screen reader users when flat on 
mobile devices. https://github.com/Financial-Times/o-table/pull/181
